src/main/java/com/avispl/symphony/dal/avdenvices/encoderdecoder/haivision/kraken/HaivisionKrakenCommunicator.java

```diff
@@ -0,0 +1,802 @@
+						if (metadataUuid.equals(metadataID)) {
+							String metadataName = metadataItem.get(HaivisionConstant.NAME).asText();
+							stats.put("Stream: " + name + HaivisionConstant.HASH + "Metadata", metadataName);
+						}
+					}
+				}
+			}
+		} catch (Exception e) {
+			logger.error("Error while populating the metadata Info", e);
+		}
+	}
+
+
+	/**
+	 * {@inheritDoc}
+   */
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) throws Exception {
+
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> controllableProperties) throws Exception {
+		if (CollectionUtils.isEmpty(controllableProperties)) {
+			throw new IllegalArgumentException("ControllableProperties can not be null or empty");
+		}
+		for (ControllableProperty p : controllableProperties) {
+			try {
+				controlProperty(p);
+			} catch (Exception e) {
+				logger.error(String.format("Error when control property %s", p.getProperty()), e);
+			}
+		}
+	}
+
+	/**
+	 * Authenticates the user by sending a login request and retrieves the session cookie or ID.
+	 *
+	 * @return {@code true} if the session is successfully retrieved, {@code false} otherwise.
+	 * @throws ResourceNotReachableException if an error occurs during the login process.
+	 */
+	private boolean getCookieSession() {
+		try {
+			HttpClient client = this.obtainHttpClient(true);
+			HttpPost httpPost = new HttpPost(buildDeviceFullPath(HaivisionCommand.API_LOGIN));
+
+			String jsonPayload = String.format(HaivisionConstant.AUTHENTICATION_PARAM, this.getLogin(), this.getPassword());
+			httpPost.setEntity(new StringEntity(jsonPayload, ContentType.APPLICATION_JSON));
+
+			HttpResponse response = client.execute(httpPost);
+			int statusCode = response.getStatusLine().getStatusCode();
+
+			if (statusCode == 403) {
+				this.authenticationCookie = EntityUtils.toString(response.getEntity());
+				return false;
+			}
+			if (statusCode == 204) {
+				Header[] headers = response.getHeaders(HaivisionConstant.SET_COOKIE);
+				if (headers.length == 0) {
+					String responseBody = EntityUtils.toString(response.getEntity());
+					String sessionId = extractSessionId(responseBody);
+					if (sessionId != null) {
+						this.authenticationCookie = "id=" + sessionId;
+						System.out.println("Session ID retrieved: " + sessionId);
+					} else {
+						System.out.println("Session ID not found in the response body.");
+						this.authenticationCookie = HaivisionConstant.EMPTY;
+						return false;
+					}
+				} else {
+					String authenticationCookies = Arrays.stream(headers)
+							.map(header -> extractUUIDFromCookie(header.getValue()))
+							.collect(Collectors.joining(""));
+
+					this.authenticationCookie = authenticationCookies.isEmpty() ? HaivisionConstant.EMPTY : authenticationCookies;
+					System.out.println("Authentication Cookies: " + this.authenticationCookie);
```

Comment:
Won't propagated to UI, but still not sure if we should print this to the logfiles (probably we can pring to logfiles, not add it to an exception message, if the cookie is already invalid - for the debugging and troubleshooting purposes, but not sure if we should do it if the cookie is still valid)